### PR TITLE
statistics: add update metrics for stats cache

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/46577

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by hi-rustin — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

File: statistics/handle/cache/statscacheinner.go [hunk 1]
```diff
@@ -118,6 +118,7 @@ func (sc *StatsCache) Put(id int64, t *statistics.Table) {
 }
 
 func (sc *StatsCache) putCache(id int64, t *statistics.Table) bool {
+	metrics.UpdateCounter.Inc()
```
Comment (qw4990) [inline statistics/handle/cache/statscacheinner.go:121]: Should we add these metrics on `LFU` instead of `StatsCache`?

Comment (hawkingrei) [reply statistics/handle/cache/statscacheinner.go:121]: In fact, MapCache will use it too.

Comment (hawkingrei) [reply statistics/handle/cache/statscacheinner.go:121]: In the future, Every cache implementation will have same metrics.

**Review by qw4990 — APPROVED** [review]

**Pull request merged**